src/main/java/com/celements/web/service/UrlScriptService.java

```diff
@@ -0,0 +1,46 @@
+package com.celements.web.service;
+
+import java.net.URL;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.xwiki.component.annotation.Component;
+import org.xwiki.component.annotation.Requirement;
+import org.xwiki.model.reference.DocumentReference;
+import org.xwiki.script.service.ScriptService;
+
+import com.celements.model.context.ModelContext;
+import com.xpn.xwiki.api.Attachment;
+
+@Component("url")
+public class UrlScriptService implements ScriptService {
+
+  private static Logger LOGGER = LoggerFactory.getLogger(UrlScriptService.class);
+
+  @Requirement
+  private ModelContext context;
+
+  @Requirement
+  private ICelementsWebServiceRole celementsweb;
+
+  public String getURL(DocumentReference docRef, String action, String queryString) {
+    URL url = context.getXWikiContext().getURLFactory().createURL(
+        docRef.getLastSpaceReference().getName(), docRef.getName(), action, queryString, null,
+        docRef.getRoot().getName(), context.getXWikiContext());
```

Comment:
`createURL` accepts a lot of parameters. for a better readability please use local variables to name them, e.g.:
`createURL(spaceName, docName, action, queryString, anchor, wikiName, context)`